diesel_derives/src/sql_function.rs

```diff
@@ -123,7 +125,55 @@ pub(crate) fn expand(input: SqlFunctionDecl) -> Result<TokenStream, Diagnostic>
         }
     };
 
-    if !is_aggregate {
+    if is_aggregate {
+        if cfg!(feature = "sqlite") && type_args.is_empty() {
+            tokens = quote! {
+                #tokens
+
+                use diesel::sqlite::{Sqlite, SqliteConnection};
+                use diesel::serialize::ToSql;
+                use diesel::deserialize::Queryable;
+                use diesel::sqlite::Aggregator;
+            };
+
+            if arg_name.len() > 1 {
+                tokens = quote! {
+                    #tokens
+
+                    #[allow(dead_code)]
+                    pub fn register_impl<A, #(#arg_name,)*>(
+                        conn: &SqliteConnection
+                    ) -> QueryResult<()>
+                        where
+                        A: Aggregator<(#(#arg_name,)*)> + Send + 'static,
+                        A::Output: ToSql<#return_type, Sqlite>,
+                        (#(#arg_name,)*): Queryable<(#(#arg_type,)*), Sqlite>,
+                    {
+                        conn.register_aggregate_function::<(#(#arg_type,)*), #return_type, _, _, A>(#sql_name)
+                    }
+                };
+            } else if arg_name.len() == 1 {
```

Comment:
I included this because I thought this:
```
fn step(&self, args: (i32,)) {
  let x = args.0;
  ...
}
```
seemed a little unintuitive when the function signature is
```
fn my_sum(expr: Integer)
```

What are your specific concerns around having this be a special case?

Reply:
My main concern is that it's an inconsistency. There is some case that is just handled differently. I see the reason why this is probably a good idea, but we need to document that accordantly. Another minor concern is that it would make it harder to write an macro that generates a similar function call for multiple tuple sizes starting by one element, but that's probably not relevant.

In that context: I just remember that we should probably update the documentation of `sql_function!` [here](https://github.com/diesel-rs/diesel/blob/5369a7a3d13b1323fcac9acf9d13256f19a970a0/diesel_derives/src/lib.rs#L558-L599) to mention support for aggregate functions and contain an simple example (just the sum test case or something like that?) Also the `Aggregator` trait needs at least a few lines of documentation pointing in principle to the `sql_function` example.

Reply:
I'll make sure to include the single and multiarg cases in the documentation :+1: